### PR TITLE
feat: add custom implementation for qwen3vlmoe 

### PR DESCRIPTION
**Pull request by HuiyingLi**
image

https://wandb.ai/Nemo-automodel/huiyingl_workspace/runs/krjmje1a

### CODE REVIEW

File: nemo_automodel/components/attention/utils.py [hunk 1]
```diff
@@ -142,8 +142,6 @@ def preprocess_args_and_kwargs_for_attn(
         k = k.transpose(1, 2).contiguous()
         v = v.transpose(1, 2).contiguous()
         attn_kwargs["is_causal"] = True
-        if attention_mask is not None and (attention_mask.dtype == torch.bool or attention_mask.is_floating_point()):
-            attn_kwargs["attn_mask"] = attention_mask
```
Comment (hemildesai) [inline nemo_automodel/components/attention/utils.py:146]: Why is this removed?

Comment (HuiyingLi) [reply nemo_automodel/components/attention/utils.py:146]: @hemildesai  Thanks for reviewing! They were added when I wanted to add a manually created causal mask for qwen3omni. Now I changed qwen3omni to right padding too, these are no longer needed. Restore back to how they were initially.
